src/Features/Core/Portable/CodeCleanup/ICodeCleanupService.cs

```diff
@@ -3,12 +3,15 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Options;
 using Microsoft.CodeAnalysis.Shared.Utilities;
 
 namespace Microsoft.CodeAnalysis.CodeCleanup
 {
     internal interface ICodeCleanupService : ILanguageService
     {
-        Task<Document> CleanupAsync(Document document, IProgressTracker progressTracker, CancellationToken cancellationToken);
+        Task<Document> CleanupAsync(Document document, EnabledDiagnosticOptions enabledDiagnostics, IProgressTracker progressTracker, CancellationToken cancellationToken);
+        EnabledDiagnosticOptions GetAllDiagnostics();
+        EnabledDiagnosticOptions GetEnabledDiagnostics(OptionSet optionSet);
```

Comment:
why do we need both of these bottom two methods?

Reply:
➡️ The second of these is getting removed in #30605

Reply:
(I started a list of these now in https://github.com/dotnet/roslyn/pull/30605#issuecomment-431413853)